phoenicis-javafx/src/main/java/org/phoenicis/javafx/controller/MainController.java

```diff
@@ -80,12 +80,21 @@ public MainController(String applicationName,
         this.themeManager = themeManager;
         this.javaFxSettingsManager = javaFxSettingsManager;
 
-        // set callbacks and ensure that they are really called at least once initially
         repositoryManager.addCallbacks(this::setDefaultCategoryIcons, e -> {
         });
-        repositoryManager.triggerCallbacks();
 
         installationsView.setOnInstallationAdded(this.mainWindow::showInstallations);
+
+        this.mainWindow.getApplicationsTab().selectedProperty().addListener((observable, oldValue, newValue) -> {
+            if (newValue && !repositoryManager.isRepositoryLoaded()) {
+                repositoryManager.triggerRepositoryChange();
+            }
+        });
+        this.mainWindow.getContainersTab().selectedProperty().addListener((observable, oldValue, newValue) -> {
+            if (newValue && !repositoryManager.isRepositoryLoaded()) {
+                repositoryManager.triggerRepositoryChange();
+            }
+        });
```

Comment:
I think you should add a comment here describing what these listeners do/should do